java-sdk/radar-schemas-tools/src/main/java/org/radarcns/schema/registration/SchemaTopicManager.java

```diff
@@ -0,0 +1,506 @@
+            Duration timeout) throws SerializationException {
+        try (Consumer<byte[], byte[]> consumer = new KafkaConsumer<>(consumerProps)) {
+            ensurePartitions(consumer);
+
+            TopicPartition topicPartition = new TopicPartition(TOPIC_NAME, 0);
+            consumer.assign(List.of(topicPartition));
+            consumer.seekToBeginning(List.of(topicPartition));
+
+            logger.debug("Kafka store reader thread started");
+
+            int numRecords = -1;
+            Duration duration = timeout;
+
+            while (numRecords != 0) {
+                ConsumerRecords<byte[], byte[]> records = consumer.poll(duration);
+                duration = SECONDARY_TIMEOUT;
+                numRecords = records.count();
+                for (ConsumerRecord<byte[], byte[]> record : records) {
+                    storeTopic.addSchemaRecord(serializer, record);
+                }
+            }
+        }
+    }
+
+    /**
+     * Ensure that _schemas has exactly one partition.
+     * @param consumer consumer to subscribe partitions with.
+     * @throws IllegalArgumentException if the _schemas topic cannot be found.
+     * @throws IllegalStateException if the _schemas topic has more than one partition.
+     */
+    private void ensurePartitions(Consumer<?, ?> consumer) {
+        // Include a few retries since topic creation may take some time to propagate and schema
+        // registry is often started immediately after creating the schemas topic.
+        int retries = 0;
+        List<PartitionInfo> partitions;
+        do {
+            partitions = consumer.partitionsFor(TOPIC_NAME);
+            if (partitions != null && !partitions.isEmpty()) {
+                break;
+            }
+
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                // ignore
+            }
+            retries++;
+        } while (retries < 10);
+
+        if (partitions == null || partitions.isEmpty()) {
+            throw new IllegalArgumentException("Unable to subscribe to the Kafka topic "
+                    + TOPIC_NAME
+                    + " backing this data store. Topic may not exist.");
+        } else if (partitions.size() > 1) {
+            throw new IllegalStateException("Unexpected number of partitions in the "
+                    + TOPIC_NAME
+                    + " topic. Expected 1 and instead got " + partitions.size());
+        }
+    }
+
+    /**
+     * Restores the _schemas from backup.
+     * @throws RuntimeException storage failure or any other error.
+     * @throws ExecutionException if the topic configuration cannot be written
+     * @throws IllegalStateException if this manager was not initialized or if the
+     *                               schema registry is running.
+     * @throws KeeperException if no connection with Zookeeper could be made.
+     */
+    public void restoreBackup(short replication)
```

Comment:
FYI: entry point to completely restore a backup. This can only be called if the `_schemas` topic was deleted.